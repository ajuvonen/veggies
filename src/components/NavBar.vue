<script setup lang="ts">
import {computed} from 'vue';
import {RouterLink, useRoute} from 'vue-router';
import HomeLocaleChanger from '@/components/HomeLocaleChanger.vue';
defineProps<{
  showStats: boolean;
}>();

const route = useRoute();

const isHome = computed(() => route.name === 'home');
</script>
<template>
  <nav :class="{'navbar--home': isHome}" class="navbar">
    <RouterLink
      v-if="!isHome"
      :aria-label="$t('views.log')"
      to="/log"
      class="nav__link nav__link--log"
      data-test-id="navbar-link-log"
      >{{ $t('general.appTitle') }}</RouterLink
    >
    <div v-if="!isHome" class="flex-container">
      <RouterLink
        v-if="showStats"
        :title="$t('views.stats')"
        :aria-label="$t('views.stats')"
        to="/stats"
        class="nav__link"
        data-test-id="navbar-link-stats"
      >
        <IconComponent icon="chart" size="6vw" class="nav__link-icon" />
      </RouterLink>
      <RouterLink
        :title="$t('views.settings')"
        :aria-label="$t('views.settings')"
        to="/settings"
        class="nav__link"
        data-test-id="navbar-link-settings"
      >
        <IconComponent icon="cog" size="6vw" class="nav__link-icon" />
      </RouterLink>
    </div>
    <HomeLocaleChanger v-else />
  </nav>
</template>
<style scoped lang="scss">
.navbar {
  @apply w-full;
  @apply flex justify-between items-center;

  &--home {
    @apply justify-end;
  }
}

.nav__link {
  @apply rounded-md leading-[1.2];
  @apply fill-slate-50;
  &:hover {
    @apply text-slate-200 fill-slate-200;
  }
}

.nav__link--log {
  font-family: 'Bungee Shade', system-ui;
  font-size: 6vw;
  @apply uppercase sm:text-4xl rounded-md;
}

.nav__link-icon {
  @apply max-h-7 min-h-5 max-w-7 min-w-5;
}
</style>
