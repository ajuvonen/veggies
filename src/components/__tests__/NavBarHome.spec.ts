import {describe, it, expect, vi} from 'vitest';
import {mount} from '@vue/test-utils';
import NavBar from '@/components/NavBar.vue';

vi.mock('vue-router', () => ({
  useRoute: () => ({
    name: 'home',
  }),
}));

describe('NavBar Home', () => {
  it('renders for home', async () => {
    const wrapper = mount(NavBar, {
      props: {
        showStats: false,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
