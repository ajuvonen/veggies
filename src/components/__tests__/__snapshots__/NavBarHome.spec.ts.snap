// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar Home > renders for home 1`] = `
"<nav data-v-c3ceb15a="" class="navbar--home navbar">
  <!--v-if-->
  <div data-v-c3ceb15a="" class="flex-container"><svg style="width: 1.25rem; height: 1.25rem;" class="flex-shrink-0" viewBox="0 0 24 24" aria-hidden="true">
      <path d="M17.9,17.39C17.64,16.59 16.89,16 16,16H15V13A1,1 0 0,0 14,12H8V10H10A1,1 0 0,0 11,9V7H13A2,2 0 0,0 15,5V4.59C17.93,5.77 20,8.64 20,12C20,14.08 19.2,15.97 17.9,17.39M11,19.93C7.05,19.44 4,16.08 4,12C4,11.38 4.08,10.78 4.21,10.21L9,15V16A2,2 0 0,0 11,18M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z" class="transition-all"></path>
    </svg><button data-v-24e38f3d="" class="button button--text" data-test-id="home-locale-button-fi" aria-label="Change language to Suomi">
      <!--v-if-->fi
    </button></div>
</nav>"
`;
